float32 goal_x
float32 goal_y
---
float32 distance
float32 avg_speed
